src/main/java/ee/tuleva/onboarding/auth/ocsp/OCSPRequest.java

```diff
@@ -0,0 +1,155 @@
+package ee.tuleva.onboarding.auth.ocsp;
+
+import com.codeborne.security.AuthenticationException;
+import lombok.extern.slf4j.Slf4j;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateHolder;
+import org.bouncycastle.cert.ocsp.*;
+import org.bouncycastle.operator.DigestCalculator;
+import org.bouncycastle.operator.DigestCalculatorProvider;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.bouncycastle.operator.jcajce.JcaDigestCalculatorProviderBuilder;
+
+import java.io.*;
+import java.math.BigInteger;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.security.Security;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+
+@Slf4j
+public class OCSPRequest {
+    private final String ocspServer;
+    private final String ca;
+    private final String method;
+
+    public OCSPRequest(String caToCheck, String ocspServer) {
+        this.ca = caToCheck;
+        this.ocspServer = ocspServer;
+        this.method = "POST";
+    }
+
+    public String checkCertificate(X509Certificate certificate){
+        if(hasCertificateExpired(certificate)){
+            return "CERTIFICATE_EXPIRED";
+        }
+        return checkSerialNumber(certificate.getSerialNumber());
+    }
+
+    private boolean hasCertificateExpired(X509Certificate certificate) {
+        return System.currentTimeMillis() > certificate.getNotAfter().getTime();
```

Comment:
Using time this way is flaky, can fail if there are timezone differences. Try to use an instance of Clock instead.